autorel/clients/github_client.py

```diff
@@ -0,0 +1,146 @@
+import base64
+import re
+from clients.http_client import HTTPClient
+from clients.github_exceptions import (BadCredentialsException,
+                                       GithubException,
+                                       UserAgentException,
+                                       ObjectNotFoundException,
+                                       UnprocessableEntityException,
+                                       TwoFactorAuthException,
+                                       LoginAttemptExceededException,
+                                       BadRequestException,
+                                       RateLimitExceededException
+                                       )
+
+
+class Github:
```

Comment:
The `Github` class itself is still operating on headers, params and payload (HTTPClient related stuff).
There is a `GithubInterface` interface instead with implementations like `PullRequests` and `Commits`.

We should talk about this architectural decision with @lbudai. :)